### PR TITLE
#3586 ui cv publish/promote with rh custom spin

### PR DESCRIPTION
**Pull request by ldjebran**
```bash
(2.7) dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/ui/test_contentview.py -v -k 'test_positive_publish_with_rh_custom_spin'
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.12, pytest-3.0.3, py-1.4.31, pluggy-0.4.0 -- /home/dlezz/bin/redhat/python/2.7/bin/python
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile: 
plugins: cov-2.4.0, xdist-1.15.0
collected 64 items 

tests/foreman/ui/test_contentview.py::ContentViewTestCase::test_positive_publish_with_rh_custom_spin <- robottelo/decorators/__init__.py PASSED

================================================= 63 tests deselected ==================================================
====================================== 1 passed, 63 deselected in 186.71 seconds =======================================
```
```bash
(2.7) dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/ui/test_contentview.py -v -k 'test_positive_promote_with_rh_custom_spin'
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.12, pytest-3.0.3, py-1.4.31, pluggy-0.4.0 -- /home/dlezz/bin/redhat/python/2.7/bin/python
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile: 
plugins: cov-2.4.0, xdist-1.15.0
collected 64 items 

tests/foreman/ui/test_contentview.py::ContentViewTestCase::test_positive_promote_with_rh_custom_spin <- robottelo/decorators/__init__.py PASSED

================================================= 63 tests deselected ==================================================
====================================== 1 passed, 63 deselected in 233.86 seconds =======================================
```

### CODE REVIEW

File: tests/foreman/ui/test_contentview.py [hunk 1]
```diff
@@ -912,11 +912,61 @@ def test_positive_promote_with_rh_custom_spin(self):
...
+            'releasever': None,
+        }
+        env_name = gen_string('alpha')
+        strategy, value = locators['content_env.select_name']
+        # Create new org to import manifest
+        org = entities.Organization().create()
+        with Session(self.browser) as session:
+            make_lifecycle_environment(
+                session, org=org.name, name=env_name)
+
```
Comment (oshtaier) [inline tests/foreman/ui/test_contentview.py]: so these empty lines before and after will be redundant

File: tests/foreman/ui/test_contentview.py [hunk 2]
```diff
@@ -912,11 +912,61 @@ def test_positive_promote_with_rh_custom_spin(self):
...
+            # add a filter
+            self.content_views.add_filter(
+                cv_name,
+                filter_name,
+                FILTER_CONTENT_TYPE['package'],
+                FILTER_TYPE['exclude'],
+            )
+            # assert the added filter visible
+            self.assertIsNotNone(self.content_views.search_filter(cv_name,
```
Comment (oshtaier) [inline tests/foreman/ui/test_contentview.py]: I would prefer:
```
self.assertIsNotNone(
    self.content_views.search_filter(cv_name, filter_name))
```

File: tests/foreman/ui/test_contentview.py [hunk 3]
```diff
@@ -912,11 +912,61 @@ def test_positive_promote_with_rh_custom_spin(self):
...
+                                                                  filter_name))
+            # exclude some package in the created filter
+            self.content_views.add_packages_to_filter(cv_name, filter_name,
+                                                      ['gofer'],
```
Comment (oshtaier) [inline tests/foreman/ui/test_contentview.py]: maybe we can put that value into constant module

File: tests/foreman/ui/test_contentview.py [hunk 4]
```diff
@@ -1046,12 +1096,61 @@ def test_positive_publish_with_rh_custom_spin(self):
...
+        filter_name = gen_string('alpha')
+        rh_repo = {
+            'name': REPOS['rhst7']['name'],
+            'product': PRDS['rhel'],
+            'reposet': REPOSET['rhst7'],
+            'basearch': 'x86_64',
+            'releasever': None,
+        }
+        env_name = gen_string('alpha')
+        strategy, value = locators['content_env.select_name']
```
Comment (oshtaier) [inline tests/foreman/ui/test_contentview.py]: Please, apply same changes for that method as I commented above

**Comment by oshtaier** [comment]
These two methods are pretty similar, but probably let's keep them as they are for better testing results tracking

**Comment by oshtaier** [comment]
ACK pending comments

File: tests/foreman/ui/test_contentview.py (hunk 3 above)
Comment (ldjebran) [inline tests/foreman/ui/test_contentview.py]: or we can totally remove that line at all,
but logically if we make a filter it should contain something to filter (some thing to include or to exclude).
this package is used for the moment only for these two tests.
Perhaps to not make a constant we can create an errata filter instead, what do you think?

File: tests/foreman/ui/test_contentview.py [hunk 5]
```diff
@@ -912,11 +912,58 @@ def test_positive_promote_with_rh_custom_spin(self):
...
+            self.content_views.add_filter(
+                cv_name,
+                filter_name,
+                FILTER_CONTENT_TYPE['package'],
+                FILTER_TYPE['exclude'],
+            )
+            # assert the added filter visible
+            self.assertIsNotNone(
+                self.content_views.search_filter(cv_name, filter_name))
+            # exclude some package in the created filter
```
Comment (oshtaier) [inline tests/foreman/ui/test_contentview.py]: exclude? or include?

Comment (oshtaier) [inline tests/foreman/ui/test_contentview.py]: ahh.. `FILTER_TYPE['exclude'],`

**Comment by oshtaier** [comment]
ACK

**Pull request merged**